RMG/4Day/로또의 최고 순위와 최저 순위.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * 자기의 로또번호와 실제 당첨 번호를 맞추어 확인하는 함수입니다.
+ * 그러나 지워져서 안 보이는 번호는 0으로 저장하고, 최소와 최대를
+ * 맞추어봅니다. (배열의 순서는 상관없습니다.)
+ * solution([44, 1, 0, 0, 31, 25],[31, 10, 45, 1, 6, 19]) // [3, 5]  -> 3등 5등
+ * solution([0, 0, 0, 0, 0, 0], [38, 19, 20, 40, 15, 25]) // [1, 6] -> 1등 6등
+ * solution([45, 4, 35, 20, 3, 9], [20, 9, 3, 45, 4, 35]) // [1, 1] -> 1등 1등
+ */
+function solution(lottos, win_nums) {
+  let mathCount = 0; // 일치하는 번호의 개수를 저장하는 변수를 선언합니다.
+  let zeroCount = 0; // 0의 개수를 저장하는 변수를 선언합니다.
+
+  // lottos 배열을 순회하며 일치하는 번호와 0의 개수를 셉니다.
+  for (let i = 0; i < lottos.length; i++) {
+    if (lottos[i] === 0) {
+      zeroCount++; // 0이면 카운트 증가
+    } else {
+      for (let j = 0; j < win_nums.length; j++) {
+        if (lottos[i] === win_nums[j]) {
+          mathCount++; // 일치하는 번호가 있으면 카운트 증가
+        }
+      }
+    }
+  }
+
+  // 최고 순위와 최저 순위를 계산합니다.
+  let highestRank, lowestRank;
+
+  // 최고 순위 계산
+  const totalMatches = mathCount + zeroCount;
+  if (totalMatches === 6) highestRank = 1;
+  else if (totalMatches === 5) highestRank = 2;
+  else if (totalMatches === 4) highestRank = 3;
+  else if (totalMatches === 3) highestRank = 4;
+  else if (totalMatches === 2) highestRank = 5;
+  else highestRank = 6; // 그 외는 낙첨
+
+  // 최저 순위 계산
+  if (mathCount === 6) lowestRank = 1;
+  else if (mathCount === 5) lowestRank = 2;
+  else if (mathCount === 4) lowestRank = 3;
+  else if (mathCount === 3) lowestRank = 4;
+  else if (mathCount === 2) lowestRank = 5;
+  else lowestRank = 6; // 그 외는 낙첨
```

Comment:
```suggestion
  // 최고 순위와 최저 순위를 계산합니다.
  let highestRank, lowestRank;

  // 최고 순위 계산
  const totalMatches = mathCount + zeroCount;
  highestRank = 7 - (totalMatches || 1);
  
  // 최저 순위 계산
  lowestRank = 7 - (mathCount || 1);
```
최고 순위와 최저 순위를 계산하는 로직을 보시면 뭔가 규칙이 보이네요.

최고 순위를 계산할 때 **totalMatches**가 감소하면 **highestRank**는 증가합니다.
마찬가지로 최저 순위를 계산할 때 **mathCount**가 감소하면 **lowestRank**는 증가합니다.
그리고 두 계산 모두 나올 수 있는 최대 수는 7이므로 다음과 같이 표현할 수 있습니다.

`최고(최저) 순위 = 7 - 일치하는 수의 개수`

그러나 문제에서 원하는 최저 순위는 최대 6위이므로 `or` 연산을 통해 0이면 1을 넣도록 구현할 수 있습니다.

Reply:
나올 수 있는 최고, 최저 순위가 매우 한정적이므로 @yihimin 희민님처럼 순위를 배열에 넣어 푸는 방법도 있습니다!

참고 : https://github.com/modern-agile-team/9term-assignment/pull/25/files#diff-4c9ce2341cf666ff55b850e8d16905bd5c177d18a86fc7911e5dfc550e4a7354

Reply:
더 깊게 생각을 못 하고 if문에 생각이 꽃혀서 다른 방법을 생각 못 했는데, 지금 보니 너무나 난잡해보이네요...
더욱 꼼꼼하게 보는 습관들 들여보겠습니다 !! 감사합니다 !!